### PR TITLE
Change schema to allow alternative configurations

### PR DESCRIPTION
**Pull request by ipetkov**
Refs https://github.com/srid/nixci/issues/2
Refs https://github.com/srid/nixci/pull/13#issuecomment-1698336854

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,12 +63,23 @@ By default, `nixci` will build the top-level flake, but you can tell it to build
...
+            dir = "dir2";
+            overrideInputs.myproject = ./.;
+        };
     };
-    dir2 = {
-        dir = "dir2";
-        overrideInputs.myproject = ./.;
+
+    # Alternative configurations can be defined and subsequently selected via
+    # `nixci --config extra-tests .`
```
Comment (srid) [inline README.md:78]: Do we really need a new option (`--config`)? Why not `nixci .#extra-tests`? 

As well, `nixci https://github.com/srid/nixci/pull/20#extra-tests`?

Comment (ipetkov) [reply README.md:78]: That's a good suggestion, I hadn't really considered it!

On first reaction I would worry whether having `nixci .#extra-tests` would be confusing (i.e. that it's really `nixci.extra-tests` in the flake) but there are other tools which do this so there's already a precedent somewhat established. Let me try taking the explicit option out

Comment (ipetkov) [reply README.md:78]: Done!

**Review by srid — CHANGES_REQUESTED** [review]
Instead of introducing a new indirection type (`FlakeRefKind`), I'd just modify the `PulRequestRef` type to take an additional struct field called `anchor`. Then, `PullRequest::flake_url` can construct a flake url that uses the anchor as attribute. 

Incidentally, this would simplify the diff even further (which is something you want to aim for in first-pass attempts).

**Comment by srid** [comment]
```rust
pub fn flake_url(&self) -> String 
```

This function can be refactored to return a `FlakeUrl` type (newtype over `String`) which then impl's a method to get the attribute.

**Comment by ipetkov** [comment]
> Instead of introducing a new indirection type (`FlakeRefKind`), I'd just modify the `PulRequestRef` type to take an additional struct field called `anchor`. Then, `PullRequest::flake_url` can construct a flake url that uses the anchor as attribute.

I've applied this feedback, let me know if you have anything else in mind!

> This function can be refactored to return a `FlakeUrl` type (newtype over `String`) which then impl's a method to get the attribute.

I did not apply this feedback because it resulted in having to wrap the url `String` in multiple places only to unwrap it later and it did not seem like the boilerplate was adding much. But if you want this refactor done I can add it in

**Comment by srid** [comment]
@ipetkov This is fine. I just realized that there is more to it than `FlakeUrl`, so I'll take over this PR and merge it to mainline.

**Pull request merged**